### ISSUE TITLE
HM-COMMENT-1

### ISSUE THREAD

**Issue by ragges** [dfoxfranke/nts#171]
HM:

> Nits.  Not worth a new version just for these, but if you are editing for some 
> other reason, please consider ...
> 
> ------------
> 
> I can't find any place where the draft explicitly says that NTS doesn't 
> guarantee anything about the quality of the time you get.  I'm concerned that 
> the "good" as applied to security will get transferred to time.  I suggest a 
> sentence in both the Introduction and Security Considerations.  Something like:
> 
> NTS doesn't say anything about the accuracy of the time you will get, only 
> that the response, if any, came from the server you expect to be talking to.
> 
> ------------
> 
> NTP_PHASE_MAX
> 
> From 9.4. Initial Verification of Server Certificates
> 
>  Check whether the system time is in fact unreliable. If the
>  system clock has previously been synchronized since last boot,
>  then on operating systems which implement a kernel-based phase-
>  locked-loop API, a call to ntp_gettime() should show a maximum
>  error less than NTP_PHASE_MAX. In this case, the clock SHOULD be
>  considered reliable and certificates can be strictly validated.
> 
> I can't find a good reference for NTP_PHASE_MAX
> It's not in RFC 5905
> It's not in the kernel sources or /usr/include/ on Linux, FreeBSD, or NetBSD.
> 
> It is in the Linux man page for ntp_gettime
>       maxerror
>              Maximum error, in microseconds.  This value can  be  initialized
>              by ntp_adjtime(3), and is increased periodically (on Linux: each
>              second), but is clamped to an upper limit (the  kernel  constant
>              NTP_PHASE_MAX, with a value of 16,000).
> 
> Google finds that man page (several translations) and the NTS draft.
> 
> I think that only works if relatively smart code set the time.  If the 
> operator set the time from his watch there probably isn't any error estimate.
> 
>

**Comment by ragges** [comment]
UW:

> Correct.
> 
> Actually even with ntpd being running, it's not really trivial to find out
> whether the clock is "well-synchronized".
>

**Comment by ragges** [comment]
Maybe we should at least reformulate this to better indicate that parts of it is Linux specific.
I think I have found NTP_PHASE_MAX before, but can't find it now (and am at the moment in a hurry, can check more later today).

**Comment by dansarie** [comment]
I'm starting to think we really have to fix this. NTP_PHASE_MAX doesn't appear to be in the Linux kernel anymore. Google searches only point to this draft and the ntp_gettime manpage. I also tried this, which didn't compile due to NTP_PHASE_MAX being an undeclared identifier.
```C
#include <sys/timex.h>
#include <stdio.h>

int main(int argc, char *argv[]) {
  printf("NTP_PHASE_MAX: %d\n", NTP_PHASE_MAX);
  return 0;
}
```

How about changing the paragraph in question to something like this?
> Check whether the system time is in fact unreliable, i.e. if it has been synchronized since last boot. If this is the case, the clock SHOULD be considered reliable and certificates can be strictly validated. The actual procedure for checking this varies between software and operating systems.

**Comment by ragges** [comment]
I think we need to change this, and that Marcus' suggestion is fine.

**Comment by dansarie** [comment]
Fixed by 1f26102683ef81f18c43c6abdc81b01f0f50b47d after discussions with Hal.
